### PR TITLE
Vampire natural armor

### LINKED ISSUE #287: No armor for vampires
Vampires and armor do not really fit together. Vampires are fast and agile creatures whereas MC armors are always clunky and heavy.

Vampires already gain a strong boost in health and strength, so they should be able to survive without armor. Therefore I am currently considering somehow preventing vampires to wear armor parts. 
Instead they could be able to wear special rings or charms which give other boosts. However, I am expecting quite some resistance against this change, so I am not sure yet.

### LINKED ISSUE #906: [Feature] Vampire cloaks feel useless
## What would you like to have changed or added to Vampirism
Vampire cloaks are outclassed by practically any chestplate in the game, so even the cost of one diamond feels not worth it. Overall, they feel like a kinda-cool-but-why feature. I heard something about being to upgrade them, but I couldn't find any indication of such a possibility existing.
Thus, I would suggest adding some kind of ability to improve a cloak - either by using it so it grows as you do or performing some kind of ritual. Upon levelling up the cloak may give more defence or some other buff like speed/jump boost/thirst rate reduction etc.

## Does this make the mod experience more balanced?
I think so. It would make a currently underpowered item into an actually useful one.

### PR DESCRIPTION
**Pull request by maxanier**
Vampires have a natural armor / tough skin. This grants them armor protection without wearing armor.
The armor value starts between leather and iron and scales up to diamond level at max vampire level.
When equipping armor anyway, the higher value (worn armor/natural armor) is applied. But the vampire speed and attack damage is reduced (except for leather armor and below).
This allows wearing armor from other mods, if they provided very strong armor that is required for gameplay.
Not sure about armor enchantments yet

#806
Closes #287
Closes #906

### CODE REVIEW

**Comment by maxanier** [comment]
TODO
- [ ] Add enchant-able vampire clothing

**Comment by Cheaterpaul** [comment]
Should the armor regeneration be cancelable by drinking milk? If not `IForgeEffect#getCurativeItems` must be overridden.

**Comment by maxanier** [comment]
Enchant-able clothing can be added later

File: src/main/java/de/teamlapen/vampirism/player/vampire/VampirePlayer.java [hunk 1]
```diff
@@ -1212,6 +1226,81 @@ private void updateFeeding() {
...
+                armorAtt.removeModifier(NATURAL_ARMOR_UUID);
+                toughnessAtt.removeModifier(NATURAL_ARMOR_TOUGHNESS_UUID);
+            }
+            else{
+                AttributeModifier modArmor = armorAtt.getModifier(NATURAL_ARMOR_UUID);
+                AttributeModifier modToughness = toughnessAtt.getModifier(NATURAL_ARMOR_TOUGHNESS_UUID);
+                double naturalArmor = getNaturalArmorValue(lvl);
+                EffectInstance armorRegen = player.getActivePotionEffect(ModEffects.armor_regeneration);
+                double armorRegenerationMod = armorRegen == null ? 0 : armorRegen.getDuration() / ((double)VampirismConfig.BALANCE.vpNaturalArmorRegenDuration.get() * 20);
+                naturalArmor *= (1-0.5*armorRegenerationMod); //Modify natural armor between 50% and 100% depending on the armor regen state
```
Comment (Cheaterpaul) [inline src/main/java/de/teamlapen/vampirism/player/vampire/VampirePlayer.java]: is 50% a little  too much? I would prefer a lower base value and a higher regeneration.

**Review by Cheaterpaul — APPROVED** [review]

File: src/main/java/de/teamlapen/vampirism/player/vampire/VampirePlayer.java (hunk 1 above)
Comment (maxanier) [reply src/main/java/de/teamlapen/vampirism/player/vampire/VampirePlayer.java]: I thought iron might be a good starting point (if you are at level 14). But yeah, it should regenerate fairly quickly, so I changed it to 25% for now

**Pull request merged**